src/master.rs

```diff
@@ -517,9 +517,97 @@ impl Master {
         Ok(time)
     }
 
+    pub fn foe_read(&mut self, idx: u16, name: String) -> Result<Vec<u8>> {
+        let file_name = string_to_foe_name(name);
```

Comment:
Hm, would it be better to fail gracefully instead of panic with a too-long string? It can be a user-supplied parameter after all.

Also reminds me that I wanted to add/copy-paste the C docstrings to all the methods here.

Reply:
Yeah I agree. None of the current `Error` variants fit though, would you rather add a new `Error::InvalidArgument` or the like or just abuse the `Io` one? The latter would look like
```rust
        let e = io::Error::new(
            io::ErrorKind::InvalidInput,
            format!(
                "FoE name can have a maximum length of 32, '{}' has {}",
                name,
                name.len()
            ),
        );
        return Err(Error::Io(e));
```

Reply:
That would be fine with me, and avoids a breaking change adding a new error enum variant. (It should probably be `non_exhaustive`...)

Reply:
`non_exhaustive` is probably a good idea, would you like me to add it?

Reply:
Unfortunately that is *also* a breaking change :)

We can do a breaking version but I'd like @flosse to weigh in there.